apps/dashboard/lib/utils.ts

```diff
@@ -22,3 +24,59 @@ export function debounce<T extends (...args: any[]) => any>(func: T, delay: numb
 
   return debounced;
 }
+
+type WorkspaceFeatures = Pick<Workspace, "features" | "betaFeatures">;
+
+type ConfigObject = WorkspaceFeatures["betaFeatures"] & WorkspaceFeatures["features"];
+
+type FlagValue<T extends keyof ConfigObject> = ConfigObject[T];
+
+/**
+ * Checks if a workspace has access to a specific feature or beta feature
+ * Returns the feature value if access is granted, undefined otherwise
+ * Note: Always returns true in development environment
+ *
+ * @param flagName - The name of the feature to check
+ * @param workspace - The workspace to check access for
+ * @returns The feature value (boolean | number | string) if access granted, undefined otherwise
+ *
+ * @example
+ * ```typescript
+ * // Check if workspace has access to logs page
+ * if (!hasWorkspaceAccess("logsPage", workspace)) {
+ *   return notFound();
+ * }
+ *
+ * // Check if workspace has access to a feature with numeric value
+ * const userLimit = hasWorkspaceAccess("userLimit", workspace);
+ * if (userLimit === undefined) {
+ *   return notFound();
+ * }
+ * ```
+ */
+export function hasWorkspaceAccess<T extends keyof ConfigObject>(
```

Comment:
nitpick, but the name is a bit weird when returning numbers like retention.

we could rename it to `flag`?
```ts
const days = flag("auditLogRetentionDays", ws)
```

Reply:
Works for me

Reply:
I did, `getFlag` instead since this is a function